qctrlopencontrols/dynamic_decoupling_sequences/predefined.py

```diff
@@ -126,40 +127,11 @@ def _add_pre_post_rotations(
     # of Y and Z pi-pulses
     preserves_11 = (y_pi_pulses + z_pi_pulses) % 2 == 0
 
-    # When states |0> and |0>+|1> are preserved, the sequence already produces
-    # an identity, so that we want the the pi/2-pulses to cancel each other out
-    if preserves_10 and preserves_11:
-        rabi_value = np.pi / 2
-        initial_azimuthal = 0
+    # use pi/2 and -pi/2 (pi/2) X pulses as pre and post rotations for CP sequences with
+    # odd(even) number of pulses
+    # always use pi/2 and -pi/2 X pulses as pre and post rotations for CPMG sequences
+    if preserves_10 or (not preserves_11):
```

Comment:
Hm, I'm not sure about this. What if we had `preserves_10=True` and `preserves_11=False` (i.e. a Z rotation)? In that case, when we had the pre (+pi/2) and post (-pi/2) rotations, we'd end up with an overall X.

Instead I think we want:
10 + 11 -> -pi/2 (overall I)
10 + ~11 -> +pi/2 (overall Z)
~10 + 11 -> +pi/2 (overall I)
~10 + ~11 -> -pi/2 (overall Z)

Reply:
Maybe we can use as the condition `(preserves_10 and preserves_11) or (not preserves_10 and not preserves_11)`

Reply:
ah, right, that is why I had a weird `extra_operation = \sigma_x * (-\sigma_z)` in the test. Thanks.